patches/roslyn/0008-Reference-version-finagling.patch

```diff
@@ -0,0 +1,299 @@
+From: Chris Rummel <crummel@microsoft.com>
+Date: Sun, 10 May 2020 22:01:06 -0500
+Subject: [PATCH 8/8] Reference version finagling
+
+---
+ eng/Versions.props                                          | 2 +-
+ .../CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj    | 6 +++++-
+ src/Compilers/CSharp/csc/csc.csproj                         | 4 ++++
+ src/Compilers/Core/Portable/Microsoft.CodeAnalysis.csproj   | 6 +++---
+ src/Compilers/Server/VBCSCompiler/VBCSCompiler.csproj       | 4 ++++
+ .../Portable/Microsoft.CodeAnalysis.VisualBasic.vbproj      | 6 +++++-
+ src/Compilers/VisualBasic/vbc/vbc.csproj                    | 4 ++++
+ .../Microsoft.CodeAnalysis.Debugging.Package.csproj         | 4 ++--
+ .../Microsoft.CodeAnalysis.PooledObjects.Package.csproj     | 4 ++--
+ .../Core/Microsoft.CodeAnalysis.EditorFeatures.csproj       | 4 ++--
+ .../Microsoft.CodeAnalysis.FunctionResolver.csproj          | 4 ++--
+ .../Microsoft.CodeAnalysis.ResultProvider.Utilities.csproj  | 2 +-
+ src/Tools/BuildBoss/BuildBoss.csproj                        | 2 +-
+ .../Microsoft.CodeAnalysis.CSharp.Workspaces.csproj         | 6 +++++-
+ .../Core/Portable/Microsoft.CodeAnalysis.Workspaces.csproj  | 4 ++++
+ .../Core/Microsoft.CodeAnalysis.Remote.Workspaces.csproj    | 4 ++--
+ .../Microsoft.CodeAnalysis.VisualBasic.Workspaces.vbproj    | 6 +++++-
+ 17 files changed, 52 insertions(+), 20 deletions(-)
+
+diff --git a/eng/Versions.props b/eng/Versions.props
+index 3b80e7eda1..1e5bb3e0c7 100644
+--- a/eng/Versions.props
++++ b/eng/Versions.props
+@@ -185,7 +185,7 @@
+     <SystemIOFileSystemPrimitivesVersion>4.3.0</SystemIOFileSystemPrimitivesVersion>
+     <SystemIOPipesAccessControlVersion>4.3.0</SystemIOPipesAccessControlVersion>
+     <SystemMemoryVersion>4.5.3</SystemMemoryVersion>
+-    <SystemRuntimeCompilerServicesUnsafeVersion>4.5.2</SystemRuntimeCompilerServicesUnsafeVersion>
++    <SystemRuntimeCompilerServicesUnsafeVersion>4.7.0</SystemRuntimeCompilerServicesUnsafeVersion>
+     <SystemRuntimeLoaderVersion>4.3.0</SystemRuntimeLoaderVersion>
+     <SystemTextEncodingCodePagesVersion>4.5.1</SystemTextEncodingCodePagesVersion>
+     <SystemTextEncodingExtensionsVersion>4.3.0</SystemTextEncodingExtensionsVersion>
+diff --git a/src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj b/src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj
+index ca1c241a5f..d1835034a6 100644
+--- a/src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj
++++ b/src/Compilers/CSharp/Portable/Microsoft.CodeAnalysis.CSharp.csproj
+@@ -75,5 +75,9 @@
+     <InternalsVisibleTo Include="Microsoft.CodeAnalysis.EditorFeatures2.UnitTests" />
+     <InternalsVisibleTo Include="CompilerBenchmarks" />
+   </ItemGroup>
++  <ItemGroup>
++    <PackageReference Include="System.Collections.Immutable" Version="1.5.0" />
++    <PackageReference Include="System.Reflection.Metadata" Version="1.6.0" />
++  </ItemGroup>
```

Comment:
Why are these being added to all (?) projects? Can this be explained in the patch, or file a new issue if it's ok to merge like this and we want to go ahead?

Reply:
Logged https://github.com/dotnet/source-build/issues/1602.